connectors/mongodb-connector/src/main/java/io/tapdata/mongodb/decoder/impl/AutoUpdateDateFilterTime.java

```diff
@@ -0,0 +1,118 @@
+package io.tapdata.mongodb.decoder.impl;
+
+import io.tapdata.entity.error.CoreException;
+import io.tapdata.mongodb.decoder.CustomSQLObject;
+import io.tapdata.util.DateUtil;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Map;
+
+public class AutoUpdateDateFilterTime implements CustomSQLObject<Object, Map<String, Object>> {
+    public static final String DYNAMIC_DATE = "$dynamicDate";
+    public static final String FORMAT = "format";
+    public static final String TO_STRING = "toString";
+    public static final String SUBTRACT = "subtract";
+
+    @Override
+    public Object execute(Object functionObj, Map<String, Object> curMap) {
+        Object filter = null;
+        boolean toString = false;
+        long subtract = 0;
+        if (functionObj instanceof Map) {
+            filter = ((Map<String, Object>) functionObj).get(FORMAT);
+            toString = Boolean.TRUE.equals(((Map<String, Object>) functionObj).get(TO_STRING));
+            Object sub = ((Map<String, Object>) functionObj).get(SUBTRACT);
+            if (sub instanceof Number) {
+                subtract = ((Number)sub).longValue();
+            } else {
+                try {
+                    subtract = Long.parseLong(String.valueOf(sub));
+                } catch (Exception ignore) {
+                    throw new IllegalArgumentException("a subtract value is illegal value, \"subtract\" must be a number");
+                }
+            }
+        }
```

Comment:
转换失败时，直接抛错。如果错误被吞，会导致结果与用户预期不一致，且无法知道原因。
抛
```suggestion
        if (functionObj instanceof Map) {
            Map<String, Object> functionMap = (Map<String, Object>) functionObj;
            filter = functionMap.get(FORMAT);
            toString = Boolean.TRUE.equals(functionMap.get(TO_STRING));
            Object sub = functionMap.get(SUBTRACT);
            if (sub instanceof Number) {
                subtract = ((Number)sub).longValue();
            } else if (null != sub) {
                throw new IllegalArgumentException("Decode '" + DYNAMIC_DATE + "' failed, '" + SUBTRACT + "' must be number: " + sub);
            }
        }
        if (null == filter) {
            throw new IllegalArgumentException("Decode '" + DYNAMIC_DATE + "' failed, '" + SUBTRACT + "' is required.");
        }
```